[Ong Han Qin] iP

hanqinilnix opened the pull request:
**Ari**

> Dear Master, what is your command? - Ari

Ari frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST

All you need to do is,

1. download it from [here](https://github.com/hanqinilnix/ip/releases/tag/A-Jar).
2. double-click it
3. add your tasks
4. let it manage your tasks for you

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

hanqinilnix committed:
Added Greetings, Echo and Exit

hanqinilnix committed:
Closed scanner

hanqinilnix committed:
Added "add" and "list" functionalities

hanqinilnix committed:
Added Task and TaskList
Added "mark" and "unmark" functionalities

hanqinilnix committed:
Included Deadline, Event and ToDo

hanqinilnix committed:
Added testing input and output

hanqinilnix committed:
Added exception and error handling

hanqinilnix committed:
Added "delete" functionality

hanqinilnix committed:
Updated input and output files

hanqinilnix committed:
Renamed Deadline, Event, ToDo to DeadlineTask, EventTask, ToDoTask

hanqinilnix committed:
Rename Duke.java to Ari.java

hanqinilnix committed:
Add "save" and "load" functionalities

hanqinilnix committed:
Modify TaskList.java

hanqinilnix committed:
Teach Ari how to understand date

hanqinilnix committed:
Modify TaskList.java

hanqinilnix committed:
Merge branch 'branch-Level-7' and 'branch-Level-8'

hanqinilnix committed:
Modify DeadlineTask.java and EventTask.java

hanqinilnix committed:
Modify everything

hanqinilnix committed:
Move all files into packages

hanqinilnix committed:
Edit package name

hanqinilnix committed:
Edit DeadlineTask.java and Parser.java

hanqinilnix committed:
Test 3 methods from 2 classes

hanqinilnix committed:
Create JAR file

hanqinilnix committed:
Remove unnecessary methods in tasks folder
Add JavaDoc

hanqinilnix committed:
Remove unnecessary constructor in storage folder
Add JavaDoc

hanqinilnix committed:
Rename DukeException.java to AriException.java
Add JavaDoc

hanqinilnix committed:
Edit JavaDoc

hanqinilnix committed:
Add hasDone method to Task.java

hanqinilnix committed:
Tweak code to comply with coding standard

hanqinilnix committed:
Add Find functionality

hanqinilnix committed:
Shift positions of some attributes

hanqinilnix committed:
Merge branch 'add-gradle-support' of https://github.com/hanqinilnix/ip into branch-A-Gradle

hanqinilnix committed:
Move test folder into src folder

hanqinilnix committed:
Change package location name

hanqinilnix committed:
Modify test cases

hanqinilnix committed:
Retrieve config from addressbook-level3

hanqinilnix committed:
Modify files with accordance to checkstyle

hanqinilnix committed:
Modify test files with accordance to checkstyle

Wei-Xinn commented on src/main/java/ari/parser/Parser.java:
```diff
@@ -0,0 +1,295 @@
+     * @throws EmptyCommandException if description is empty
+     */
+    private String getArgument(String command) throws EmptyCommandException {
+        String[] cmdArray = command.split(" ");
```
```suggestion
        String[] commandArray= command.split(" ");
```
Consider changing "cmd" to "command" as it seems like that is the convention being used.

Wei-Xinn commented on src/main/java/ari/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class DeadlineTask extends Task {
+    protected String time;
+    protected LocalDate nowTime;
```
Should a more intuitive variable name be used? Maybe something like deadlineDate

Wei-Xinn commented on src/main/java/ari/tasks/EventTask.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class EventTask extends Task {
+    protected String time;
+    protected LocalDate nowTime;
```
Should a more intuitive variable name be used? Maybe something like eventDate

Wei-Xinn commented on src/main/java/ari/ui/Ui.java:
```diff
@@ -0,0 +1,72 @@
+ * Deals with the interactions with the user
+ */
+public class Ui {
+    private static final String LINE_BREAK = "____________________________________________________________";
+    private static final String LOGO = "             ___      .______       __         \n"
+            + "            /   \\     |   _  \\     |  |      \n"
+            + "           /  ^  \\    |  |_)  |    |  |      \n"
+            + "          /  /_\\  \\   |      /     |  |      \n"
+            + "         /  _____  \\  |  |\\  \\----.|  |      \n"
+            + "        /__/     \\__\\ | _| `._____||__|      \n"
+            + "\n";
+    private static final String WELCOME_MESSAGE = "Dear Master, I am Ari, your Personal Assistant Chatbot";
+    private static final String WELCOME_QUESTION = "What can I do for you?";
```
I like this naming convention. It makes its usage clear and convenient!

Wei-Xinn commented on src/main/java/ari/command/FindCommand.java:
```diff
@@ -0,0 +1,50 @@
+ * Finds Task in TaskList that contains the keyword
+ */
+public class FindCommand extends Command {
+    public static final String COMMAND_WORD = "find";
+
+    private static final String FIND_MESSAGE = "Yes Master, here are the tasks in your list that matches the keyword:";
+    private static final String NOT_FOUND_MESSAGE = "I am sorry Master, "
+            + "the keyword you are searching for is not in your list";
+    private static final String EMPTY_LIST_MESSAGE = "Dear Master, you have not added anything";
+    private static final String INDEX_MESSAGE = "%2d. ";
```
Good naming convention! Makes variable usage neat!

Wei-Xinn commented on src/main/java/ari/parser/Parser.java:
```diff
@@ -0,0 +1,295 @@
+     */
+    public String[] fileBreakdown(String fileCommand) {
+        String[] words = fileCommand.split(" ");
+        String[] returnCommands = new String[3];
```
```suggestion
        String[] commandArguments= new String[3];
```
Is the variable name returnCommands unintuitive? Maybe something like commandArugments would make its purpose clearer

Wei-Xinn commented on src/main/java/ari/Ari.java:
```diff
@@ -0,0 +1,52 @@
+    public void run() {
+        ui.displayWelcomeMessage();
+
+        Parser par = new Parser();
```
```suggestion
        Parser parser = new Parser();
```
Consider changing the naming variable to match the convention. Since the Ui variable is named ui and the Storage variable is named storage.

hanqinilnix committed:
Move Main.java into ari

hanqinilnix committed:
Add GUI functionality

hanqinilnix committed:
Update src/main/java/ari/parser/Parser.java

Change variable names

Co-authored-by: Wei-Xinn <97611307+Wei-Xinn@users.noreply.github.com>

hanqinilnix committed:
Update src/main/java/ari/parser/Parser.java

Change variable name

Co-authored-by: Wei-Xinn <97611307+Wei-Xinn@users.noreply.github.com>

hanqinilnix committed:
Change archive base name

hanqinilnix committed:
Change variable names

hanqinilnix committed:
Add ViewSchedule functionality

This functionality provides a way to view tasks in the form
of a schedule.

We can have a simpler view of the schedule of the tasks of a
specific date

hanqinilnix committed:
Improve code quality

Critically examined the code and refactor to improve the code quality
where necessary

hanqinilnix committed:
Add assert statements

The assert statements help to ensure that the program is running
correctly at the point in the code.

hanqinilnix committed:
Merge pull request #1 from hanqinilnix/branch-Assertions

Branch assertions

hanqinilnix committed:
Merge branch 'master' into branch-CodingQuality

hanqinilnix committed:
Merge pull request #2 from hanqinilnix/branch-CodingQuality

Branch coding quality

hanqinilnix committed:
Merge branch 'master' into branch-Extension

hanqinilnix committed:
Merge pull request #3 from hanqinilnix/branch-Extension

Add ViewSchedule functionality

hanqinilnix committed:
Move DialogBox and MainWindow into gui folder

hanqinilnix committed:
Add functionality to close GUI when ByeCommand is executed

hanqinilnix committed:
Set theme jekyll-theme-slate

hanqinilnix committed:
Add User Guide

hanqinilnix committed:
Merge branch 'master' of https://github.com/hanqinilnix/ip

hanqinilnix committed:
Add welcome message

hanqinilnix committed:
Remove literal messages

hanqinilnix committed:
Remove used methods
